ftp/src/test/scala/akka/stream/alpakka/ftp/CommonFtpStageSpec.scala

```diff
@@ -243,5 +244,21 @@ trait CommonFtpStageSpec extends BaseSpec {
 
       result.status.failed.get shouldBe a[ArithmeticException]
     }
+
+    "fail and report the exception in the result status if connection fails" in {
+      val fileName = "sample_io"
+      val infiniteSource = Source.repeat(ByteString(0x00))
+
+      val future = infiniteSource.runWith(storeToPath(s"/$fileName", append = false))
+      eventually {
+        noException should be thrownBy getFtpFileContents(FtpBaseSupport.FTP_ROOT_DIR, fileName)
+        getFtpFileContents(FtpBaseSupport.FTP_ROOT_DIR, fileName).length shouldBe >(0)
+      }
+      stopServer()
```

Comment:
OK, so this simulates stopping the server while writing is in progress. Not super easy to read but a nice 'integration'-style test.